src/postgres/postgresConnectionStrings.ts

```diff
@@ -32,16 +37,21 @@ export function fixedEncodeURIComponent(component: string): string {
 
 export class ParsedPostgresConnectionString extends ParsedConnectionString {
     public readonly hostName: string;
-    public username: string | undefined;
-    public password: string | undefined;
+    public username: string;
```

Comment:
Even for localhost,  username is required but password can remain empty. And the parse() function assumes values for username and password. And same goes for ClientConfig. 

`export interface ConnectionOptions {
  host: string | null
  password?: string
  user?: string
  port?: string | null
  database: string | null | undefined
  client_encoding?: string
  ssl?: boolean | string`

Reply:
In the type that you copied, "user" has a question mark, meaning it's optional

For more detail on why these are optional, I took a look at these docs: https://www.postgresql.org/docs/10/libpq-connect.html

- password: "Password to be used if the server demands password authentication."
    - The key word here is "if". We don't know which cases require a password and which don't, so we need to support both (Plus - as mentioned before - localhost is the most important attached scenario and it doesn't need a password)
- user: "PostgreSQL user name to connect as. Defaults to be the same as the operating system name of the user running the application."
    - Meaning "postgres" is _not_ the default in all scenarios. Also means we don't need to give a username because the server will determine a default by itself

Reply:
That makes sense. So we are assuming then that users will provide both the ssl and credentials with the connection string?

Reply:
Yes - it's up to the user to provide (or not provide) ssl and credentials